bee/src/main/java/com/apisense/bee/services/EventObserver.java

```diff
@@ -0,0 +1,170 @@
+package com.apisense.bee.services;
+
+import android.os.SystemClock;
+import android.util.Log;
+import android.view.accessibility.AccessibilityEvent;
+
+import com.apisense.bee.utils.accessibilitySting.AccessibilityEventWrapper;
+
+import java.util.Stack;
+
+import io.apisense.sting.phone.system.WindowChangeDetectingService;
+
+/**
+ * Created by Mohammad Naseri
+ */
+
+public class EventObserver extends WindowChangeDetectingService {
+    private static final String TAG = EventObserver.class.getName();
+
+    private static OnAccessibilityEvent callback;
+    private static Stack<StackElement> events = new Stack<>();
+    private static StackElement lastEvent = null;
+    private static String tempInput = "";
+    private static AccessibilityEventWrapper tempWrapper;
+
+    private static String EDIT_TEXT_CLASS = "android.widget.EditText";
+    private static String WIDGET_CLASS = "android.widget";
+
+    @Override
+    public void onServiceConnected() {
+        Log.d(TAG, "onServiceConnected");
+
+        long t_start = SystemClock.uptimeMillis();
+        Log.d(TAG, "EXECUTION START-TIME: " + t_start);
+    }
+
+    @Override
+    public void onAccessibilityEvent(AccessibilityEvent event) {
+        super.onAccessibilityEvent(event);
+
+
+        if (events.size() > 0)
+            lastEvent = events.lastElement();
+
+        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_TEXT_SELECTION_CHANGED &&
+                (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED))
+                        || (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_VIEW_TEXT_CHANGED))
+                ))) {
+            if (event.getText().size() != 0) {
+                if (event.isPassword()) {
+                    tempInput += getLastCharacter(String.valueOf(event.getText().get(0)));
+                    tempWrapper = new AccessibilityEventWrapper(event, tempInput);
+                } else {
+                    tempInput = String.valueOf(event.getText().get(0));
+                    tempWrapper = new AccessibilityEventWrapper(event, tempInput);
+                }
+            }
+        }
+
+        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_FOCUSED &&
+                (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_VIEW_TEXT_SELECTION_CHANGED))
+                        || (lastEvent.eventType.equals(AccessibilityEvent.eventTypeToString(AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED))))) {
+
+            if (!tempInput.isEmpty()) {
+//                    Log.d(TAG, "onAccessibilityEvent: " + tempWrapper.toString());
```

Comment:
Don't keep commented  code, you can either:

1. Remove the useless lines
1. Set them back with a useful message either in _debug_ or _verbose_

Reply:
the thing is that I don't want to log the sensitive data in the production mode, and also I need those for in debug mode. Anyway, I removed them for now.

Reply:
Our proguard configuration will remove the _debug_ and _verbose_ log from release,
so you can keep them.